source/multigrid/mg_transfer_prebuilt.cc

```diff
@@ -241,6 +269,20 @@ void MGTransferPrebuilt<VectorType>::build_matrices
           {
             cell->get_mg_dof_indices (dof_indices_parent);
 
+            // Replace any indices of identity constrained dofs with the
+            // indices of the dofs in which they are constrained
+            if (this->mg_constrained_dofs != nullptr)
```

Comment:
the block of code here is repeated 6 times. Maybe move it into a helper function in some anonymous namespace that gets `this->mg_constrained_dofs->get_level_constraint_matrix(level)` (or with `level+1` or `level-1`) and `dof_indices` (or `dof_indices_parent` or `dof_indices_child`) as arguments?